### ISSUE TITLE
Kargo UI always centers and has slow scroll speed

### ISSUE THREAD

**Issue by moro-drake** [akuity/kargo#3161]
# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate bug report.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `kargo version`.

# Description

With the release of v1.1.1 the UI now centers on project space. This is good when the project does not involve too many stages, but when it's bigger - that combined with the slow scroll speed, make it impossible to interact with things that are far out from center.
With a structure like on the screenshot kargo will always center to the middle 'blue' Stage, making it really hard to interact with Warehouses and Stages that part of the  first and last pipeline in the attached picture.

I tested this on Windows PC chrome and edge, asked my colleagues to check it on their Mac - it is the same.

# Screenshots

![image](https://github.com/user-attachments/assets/30a8702d-9f2f-4c93-9068-4406aa21efee)

# Steps to Reproduce

1. Create a project with 5 separate Warehouses and stages, ideally with 2 or more subscriptions to each Warehouse
2. Inspect the UI - the scroll is slow and zooming in / out will always make it center to the 3rd 'pipeline' in the project.

# Version
v1.1.1

**Comment by krancour** [comment]
This was on our radar, but we don't seem to have had an issue for it. Thanks for opening one!

**Comment by aayushsrivastava** [comment]
In case we are open to suggestions...

Scrolling across the graph is a more common action. Zoom is something you adjust once in a while. Therefore, ideally the scrolling should require _lesser effort_ from the end-user than adjusting the zoom level.

Right now, we use vertical scroll for zoom and click-and-drag for scrolling across the graph. 

I think we should use vertical+horizontal scroll for scrolling the graph. For zoom, we can do a pinch and/or CMD+Vertical scroll. **Excalidraw** is probably a good source of inspiration for what I just described.

**Comment by krancour** [comment]
@aayushsrivastava I agree. I think I frequently get tripped up by using the gestures I'm accustomed to for scrolling and finding it adjusts zoom level instead.

@Marvin9 what do you think about this suggestion?

**Comment by Marvin9** [comment]
I agree with the suggestion and it was originally intended to work like that when I did pipeline view improvements. But I am now seeing the limitation of our custom graph UI and issues keep piling up for busy pipeline view. I will open up new issue for longer term solution

**Comment by Marvin9** [comment]
fixed as part of https://github.com/akuity/kargo/issues/3204
